src/JsonSchema/Uri/UriRetriever.php

```diff
@@ -65,9 +83,10 @@ public function confirmMediaType($uriRetriever, $uri)
             return;
         }
 
-        if (substr($uri, 0, 23) == 'http://json-schema.org/') {
-            //HACK; they deliver broken content types
-            return true;
+        for ($i = 0, $iMax = count($this->mediaTypeBlacklist); $i < $iMax; $i++) {
+            if (stripos($uri, $this->mediaTypeBlacklist[$i]) === 0) {
```

Comment:
Should be case-sensitive.

Reply:
@erayd Can you elaborate? Which part needs to be case-sensitive?

Reply:
@estahn Most of a URL is case-sensitive; only the domain and scheme are not. As this is a feature which does prefix matching on a full URL, we should be checking case. So `stripos()` is not appropriate here.

An alternative would be to parse the URL properly and separate out the origin, if you would like to distinguish between the sensitive and insensitive parts.

Reply:
@erayd I see. Would you consider something like `strtolower` is ok, given it's more a theoretical issue?

Reply:
@estahn No, I do not consider that to be OK. It's not just a theoretical issue; this has practical implications (it's a security problem, and this library is widely used). `strtolower` doesn't solve anything, it just transfers the case-insensitivity to a different place.

Reply:
@estahn Yes, using `strpos` is OK :-).